src/main.rs

```diff
@@ -602,12 +602,15 @@ async fn main() -> Result<()> {
         }
     }
 
-    let uuid = match config.agent.uuid.as_str() {
+    // Replace the uuid with the actual EK hash if the option was set.
+    // We cannot do that when the configuration is loaded initially,
+    // because only have later access to the the TPM.
+    config.agent.uuid = match config.agent.uuid.as_str() {
```

Comment:
Makes a lot of sense, but I would move it a bit early, around line 509 where `ek_hash` is already calculated just after `ek_result` is created. This will also remove some duplication.

Reply:
Good catch, I moved the change closer to line 509 and use the hash_ek value.